flow/layers/opacity_layer.cc

```diff
@@ -77,7 +79,7 @@ void OpacityLayer::Paint(PaintContext& context) const {
       context.leaf_nodes_canvas->getTotalMatrix()));
 #endif
 
-  if (context.raster_cache &&
+  if (context.raster_cache && ShouldRasterCache() &&
```

Comment:
If you didn't prepare it above (due to Should) then it might still be in the cache and taking up memory. Not using it here would be a waste.

On the other hand, if you use it here then it will be marked as "touched this frame" and it will not be culled at the end of this frame. (Which then means it still exists on the next frame to potentially be reused even if alpha is 255).

We don't have a way to delete cache entries that weren't "prepared" on a given frame between Preroll and Paint().

So, calling Should() here will eventually allow it to be reclaimed, but it will happen after you ignored it here and painted the children manually meaning it is wasted space for one frame. On the other hand, not calling Should() here would mean that it would get touched this frame and survive to the next - and so on and so on. It would be nice to say "draw it from the cache if you have it, but don't touch it", but we don't have that function.